File: Python3/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    left, right = 0,0
    total = sequence[0]
    while right < len(sequence):
        if total == k:
            if not answer or (right-left) < (answer[1] - answer[0]):
                answer = [left,right]
            total -= sequence[left]
            left += 1
        elif total < k:
            right += 1
            if right < len(sequence):
                total += sequence[right]
        else:
            total -= sequence[left]
            left += 1
    return answer

File: Java/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int len = sequence.length;
        int[] answer = {0,len};
        int start = 0;
        int end = 0;
        int total = sequence[0];
        
        while(end < len) {
            if(total == k){
                if(answer[1] - answer[0] > end - start){
                    answer[0] = start;
                    answer[1] = end;
                }
                
                total -= sequence[start];
                start ++;
            } else if(total < k) {
                end ++;
                if(end < len)
                    total += sequence[end];
            } else {
                total -= sequence[start];
                start ++;
            }
        }
        return answer;
    }
}